client/jetpack-connect/utils.js

```diff
@@ -148,7 +148,7 @@ export function parseAuthorizationQuery( query ) {
  * @param  {?Object}    queryArgs Query parameters
  * @returns {string}        Redirect url
  */
-export function redirect( type, url, product = null, queryArgs = {} ) {
+export function redirect( type, url, product = null, queryArgs = {}, source = null ) {
```

Comment:
I wonder if we need a separate `source` argument here. Why can't we just pass it via the `queryArgs`?

Reply:
Agree. I have made the changes.